File: components/ProfilePage.js
import React, { useState } from 'react';

const ProfilePage = () => {
    const [userInfo, setUserInfo] = useState({
        name: 'John Doe',
        email: 'john.doe@example.com',
        phone: '123-456-7890',
    });

    const [orderHistory, setOrderHistory] = useState([
        { id: 1, date: '2023-10-01', items: 'Pizza, Coke', total: '$20', status: 'Delivered' },
        { id: 2, date: '2023-09-25', items: 'Burger, Fries', total: '$15', status: 'Delivered' },
    ]);

    const [paymentMethods, setPaymentMethods] = useState([
        { id: 1, type: 'Credit Card', last4: '1234' },
        { id: 2, type: 'PayPal', email: 'john.paypal@example.com' },
    ]);

    const handleLogout = () => {
        console.log('User logged out');
        // Add logout logic here
    };

    const handleUpdateProfile = (e) => {
        e.preventDefault();
        console.log('Profile updated:', userInfo);
        // Add update logic here
    };

    return (
        <div>
            <h1>Profile Page</h1>

            {/* Editable Profile Information */}
            <section>
                <h2>Edit Profile</h2>
                <form onSubmit={handleUpdateProfile}>
                    <label>
                        Name:
                        <input
                            type="text"
                            value={userInfo.name}
                            onChange={(e) => setUserInfo({ ...userInfo, name: e.target.value })}
                        />
                    </label>
                    <label>
                        Email:
                        <input
                            type="email"
                            value={userInfo.email}
                            onChange={(e) => setUserInfo({ ...userInfo, email: e.target.value })}
                        />
                    </label>
                    <label>
                        Phone:
                        <input
                            type="tel"
                            value={userInfo.phone}
                            onChange={(e) => setUserInfo({ ...userInfo, phone: e.target.value })}
                        />
                    </label>
                    <button type="submit">Save Changes</button>
                </form>
            </section>

            {/* Order History */}
            <section>
                <h2>Order History</h2>
                <ul>
                    {orderHistory.map((order) => (
                        <li key={order.id}>
                            {order.date} - {order.items} - {order.total} - {order.status}
                        </li>
                    ))}
                </ul>
            </section>

            {/* Payment Methods */}
            <section>
                <h2>Payment Methods</h2>
                <ul>
                    {paymentMethods.map((method) => (
                        <li key={method.id}>
                            {method.type} - {method.last4 || method.email}
                        </li>
                    ))}
                </ul>
            </section>

            {/* Logout Option */}
            <section>
                <button onClick={handleLogout}>Logout</button>
            </section>
        </div>
    );
};

export default ProfilePage;

File: lib/screens/profile_page.dart
import 'package:flutter/material.dart';

class ProfilePage extends StatefulWidget {
  const ProfilePage({Key? key}) : super(key: key);
  
  @override
  _ProfilePageState createState() => _ProfilePageState();
}

class _ProfilePageState extends State<ProfilePage> {
  final TextEditingController _nameController = TextEditingController(text: "John Doe");
  final TextEditingController _emailController = TextEditingController(text: "john.doe@example.com");
  final TextEditingController _phoneController = TextEditingController(text: "123-456-7890");

  final List<Map<String, String>> _orderHistory = [
    {"date": "2025-04-20", "items": "Pizza, Coke", "total": "\$20", "status": "Delivered"},
    {"date": "2025-04-18", "items": "Burger, Fries", "total": "\$15", "status": "Delivered"},
  ];

  final List<Map<String, String>> _paymentMethods = [
    {"type": "Visa **** 1234", "details": "12/23"},
    {"type": "PayPal", "details": "john.doe@example.com"},
  ];

  void _saveProfile() {
    // Add save profile logic here
    print("Profile updated: ${_nameController.text}, ${_emailController.text}, ${_phoneController.text}");
  }

  void _logout() {
    // Add logout logic here
    print("User logged out");
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("Profile"),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text("Edit Profile", 
                style: Theme.of(context).textTheme.titleLarge?.copyWith(
                  fontWeight: FontWeight.bold
                )),
            const SizedBox(height: 16),
            
            TextField(
              controller: _nameController,
              decoration: const InputDecoration(
                labelText: "Name",
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 12),
            
            TextField(
              controller: _emailController,
              decoration: const InputDecoration(
                labelText: "Email",
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 12),
            
            TextField(
              controller: _phoneController,
              decoration: const InputDecoration(
                labelText: "Phone",
                border: OutlineInputBorder(),
              ),
            ),
            const SizedBox(height: 20),
            
            ElevatedButton(
              onPressed: _saveProfile,
              child: const Text("Save Changes"),
            ),
            const Divider(height: 32),

            Text("Order History", 
                style: Theme.of(context).textTheme.titleLarge?.copyWith(
                  fontWeight: FontWeight.bold
                )),
            const SizedBox(height: 8),
            ..._orderHistory.map((order) => Card(
              child: ListTile(
                title: Text(order["items"]!),
                subtitle: Text("${order["date"]} - ${order["status"]}"),
                trailing: Text(order["total"]!),
              ),
            )).toList(),
            const Divider(height: 32),

            Text("Payment Methods", 
                style: Theme.of(context).textTheme.titleLarge?.copyWith(
                  fontWeight: FontWeight.bold
                )),
            const SizedBox(height: 8),
            ..._paymentMethods.map((method) => Card(
              child: ListTile(
                title: Text(method["type"]!),
                subtitle: Text(method["details"]!),
              ),
            )).toList(),
            const Divider(height: 32),

            Center(
              child: ElevatedButton(
                onPressed: _logout,
                style: ElevatedButton.styleFrom(
                  backgroundColor: Colors.red,
                  padding: const EdgeInsets.symmetric(horizontal: 32, vertical: 12),
                ),
                child: const Text("Logout"),
              ),
            ),
            const SizedBox(height: 20),
          ],
        ),
      ),
    );
  }
}
